apps/trading/lib/hooks/use-position-closeout-notification.tsx

```diff
@@ -0,0 +1,327 @@
+import { useApolloClient } from '@apollo/client';
+import {
+  PositionsDocument,
+  type PositionsQuery,
+  type PositionsQueryVariables,
+} from '@vegaprotocol/positions';
+import {
+  TradesDocument,
+  TradesUpdateDocument,
+  type TradeUpdateFieldsFragment,
+  type TradesUpdateSubscription,
+  type TradesQuery,
+  type TradesQueryVariables,
+  type TradesUpdateSubscriptionVariables,
+  type TradeFieldsFragment,
+} from '@vegaprotocol/trades';
+import { Side, TradeType } from '@vegaprotocol/types';
+import {
+  Intent,
+  type Toast,
+  useToasts,
+  ToastHeading,
+  createActiveTabKey,
+} from '@vegaprotocol/ui-toolkit';
+import { addDecimalsFormatNumber } from '@vegaprotocol/utils';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import { useCallback, useEffect, useState } from 'react';
+import { useT } from '../use-t';
+import { useMarketsMapProvider } from '@vegaprotocol/markets';
+import { type AssetFieldsFragment } from '@vegaprotocol/assets';
+import { Link } from 'react-router-dom';
+import { Routes } from '../links';
+import { useLocalStorage } from '@vegaprotocol/react-helpers';
+import uniq from 'lodash/uniq';
+
+import { usePositionsStore } from '../../components/positions-container';
+import { PORTFOLIO_TOP_TABS } from '../../client-pages/portfolio/portfolio';
+import { theme } from '@vegaprotocol/tailwindcss-config';
+
+const LOCAL_STORAGE_KEY = 'vega_position_close_out_notifications';
+
+const createToastId = (tradeId: string) =>
+  `position_close_out_notification_${tradeId}`;
+
+const parseLocalStorageValue = (value: string | null | undefined): string[] => {
+  try {
+    if (value) {
+      const json = JSON.parse(value);
+      if (Array.isArray(json)) {
+        return json.filter((v) => v.match(/^[0-9a-f]{64}$/));
+      }
+    }
+  } catch {
+    // NOOP
+  }
+  return [];
+};
+
+const useSeen = () => {
+  const [localStorageValue, setLocalStorageValue] =
+    useLocalStorage(LOCAL_STORAGE_KEY);
+  const seen = parseLocalStorageValue(localStorageValue);
+  const setSeen = useCallback(
+    (tradeId: string) => {
+      const newSeen = uniq([...(seen || []), tradeId]);
+      setLocalStorageValue(JSON.stringify(newSeen));
+    },
+    [seen, setLocalStorageValue]
+  );
+
+  return { seen, setSeen };
+};
+
+const PositionClosedOutToastContent = ({
+  closeOutPrice,
+  asset,
+  marketId,
+  partyId,
+  onClose,
+}: {
+  closeOutPrice: string;
+  asset: Pick<AssetFieldsFragment, 'symbol'>;
+  marketId: string;
+  partyId: string;
+  onClose?: () => void;
+}) => {
+  const t = useT();
+  const [value, setValue] = useLocalStorage(
+    createActiveTabKey(PORTFOLIO_TOP_TABS)
+  );
+  const showClosedPositions = usePositionsStore(
+    (state) => state.showClosedMarkets
+  );
+
+  const highlightPositionElement = () => {
+    let tries = 0;
+    const int = setInterval(() => {
+      tries += 1;
+      if (tries > 10) {
+        clearInterval(int);
+      }
+      const position = document.querySelector(
```

Comment:
Dont like this. If we move to a persisted zustand store we can access it in the positions table